src/runtime/propensity-server.js

```diff
@@ -175,6 +176,53 @@ export class PropensityServer {
     );
   }
 
+  /**
+   * @param {string} url
+   * @private
+   */
+  sendBeacon_(url) {
+    this.setupGDPR_();
+    this.tcDataPromise_.then(tcdata => {
+      if (tcdata) {
```

Comment:
If TCData never becomes available (and GDPR is setup) then the promise result will never get called.  Is this OK?